packages/components/src/internal/components/domainproperties/dataset/DatasetDesignerPanels.tsx

```diff
@@ -316,6 +323,34 @@ export class DatasetDesignerPanelImpl extends React.PureComponent<Props & Inject
         );
     };
 
+    onImportErrorStayAndFix = () => {
+        const { model } = this.state;
+
+        this.setState({importErrorModalOpen: false});
+        const dropOptions = {
+            schemaName: 'study',
+            queryName: model.name,
+            failure: (error) => {
+                this.setState(
+                    produce((draft: Draft<State>) => {
+                        draft.model.exception = error;
+                }))
+            },
+            success: () => {
+                this.setState(produce((draft: Draft<State>) => {
+                    draft.model.domain = this.state.preSaveDomain;
+                }), () => {
+                    this.setState({preSaveDomain: undefined});
```

Comment:
can this just be set in the `this.setState()` from line 340, rather then calling this.setState() again as a callback? I believe it would just be adding in `draft.preSaveDomain = undefined`.

Reply:
Done!